pkg/detectors/gitlab/v3/gitlab_v3.go

```diff
@@ -0,0 +1,101 @@
+package gitlab
+
+import (
+	"context"
+	"fmt"
+	"maps"
+	"net/http"
+	"strings"
+
+	regexp "github.com/wasilibs/go-re2"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	v1 "github.com/trufflesecurity/trufflehog/v3/pkg/detectors/gitlab/v1"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+	detectors.EndpointSetter
+}
+
+// Ensure the Scanner satisfies the interfaces at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+var _ detectors.EndpointCustomizer = (*Scanner)(nil)
+var _ detectors.Versioner = (*Scanner)(nil)
+
+func (Scanner) Version() int          { return 3 }
+func (Scanner) CloudEndpoint() string { return "https://gitlab.com" }
+
+var (
+	defaultClient = common.SaneHttpClient()
+	keyPat        = regexp.MustCompile(`\b(glpat-[a-zA-Z0-9\-=_]{27,300}.[0-9a-z]{2}.[a-z0-9]{9})\b`)
```

Comment:
__optional__: It's probably worth adding a comment and a link to [the gitlab commit](https://gitlab.com/gitlab-org/gitlab/-/merge_requests/169322) that sets the regex, just so anyone curious about it can more easily pull up that info.

Though, it's also included as a part of the PR, so the information _is_ available, which is great! A comment would just make it easier.